internal/services/dataprotection/data_protection_backup_policy_postgresql_flexible_server_resource.go

```diff
@@ -0,0 +1,700 @@
+				Name:    id.BackupPolicyName,
+				VaultId: vaultId.ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				if properties, ok := model.Properties.(backuppolicies.BackupPolicy); ok {
+					state.DefaultRetentionRule = flattenBackupPolicyPostgreSQLFlexibleServerDefaultRetentionRule(properties.PolicyRules)
+					state.RetentionRules = flattenBackupPolicyPostgreSQLFlexibleServerRetentionRules(properties.PolicyRules)
+					state.BackupRepeatingTimeIntervals = flattenBackupPolicyPostgreSQLFlexibleServerBackupRules(properties.PolicyRules)
+					state.TimeZone = flattenBackupPolicyPostgreSQLFlexibleServerBackupTimeZone(properties.PolicyRules)
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r DataProtectionBackupPolicyPostgreSQLFlexibleServerResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.DataProtection.BackupPolicyClient
+
+			id, err := backuppolicies.ParseBackupPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if _, err := client.Delete(ctx, pointer.From(id)); err != nil {
+				return fmt.Errorf("deleting %s: %+v", pointer.From(id), err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandBackupPolicyPostgreSQLFlexibleServerAzureBackupRules(input []string, timeZone string, taggingCriteria []backuppolicies.TaggingCriteria) []backuppolicies.BasePolicyRule {
+	results := make([]backuppolicies.BasePolicyRule, 0)
+
+	results = append(results, backuppolicies.AzureBackupRule{
+		Name: "BackupIntervals",
+		DataStore: backuppolicies.DataStoreInfoBase{
+			DataStoreType: backuppolicies.DataStoreTypesVaultStore,
+			ObjectType:    "DataStoreInfoBase",
+		},
+		BackupParameters: backuppolicies.AzureBackupParams{
+			BackupType: "Full",
+		},
+		Trigger: backuppolicies.ScheduleBasedTriggerContext{
+			Schedule: backuppolicies.BackupSchedule{
+				RepeatingTimeIntervals: input,
+				TimeZone:               pointer.To(timeZone),
+			},
+			TaggingCriteria: taggingCriteria,
+		},
+	})
+
+	return results
+}
+
+func expandBackupPolicyPostgreSQLFlexibleServerAzureRetentionRules(input []BackupPolicyPostgreSQLFlexibleServerRetentionRule) []backuppolicies.BasePolicyRule {
+	results := make([]backuppolicies.BasePolicyRule, 0)
+
+	for _, item := range input {
+		results = append(results, backuppolicies.AzureRetentionRule{
+			Name:       item.Name,
+			IsDefault:  utils.Bool(false),
+			Lifecycles: expandBackupPolicyPostgreSQLFlexibleServerLifeCycle(item.LifeCycle),
+		})
+	}
+
+	return results
+}
+
+func expandBackupPolicyPostgreSQLFlexibleServerDefaultAzureRetentionRule(input []BackupPolicyPostgreSQLFlexibleServerDefaultRetentionRule) backuppolicies.BasePolicyRule {
+	return backuppolicies.AzureRetentionRule{
+		Name:       "Default",
+		IsDefault:  utils.Bool(true),
+		Lifecycles: expandBackupPolicyPostgreSQLFlexibleServerLifeCycle(input[0].LifeCycle),
```

Comment:
It's required property so that it wouldn't occurs this situation.

Reply:
I think it's still necessary because we should make sure the method will always work properly without any crash no matter whether it's a required property or not.

Reply:
Added empty check.